File: Java/src/covariance/informatics/CovNetwork.java
package covariance.informatics;

import covariance.abs.AbsEdge;
import covariance.abs.AbsGraph;
import covariance.abs.AbsVertex;

public class CovNetwork extends AbsGraph
{

    public CovNetwork(String name)
    {
        super("Network " + name);
    }

    public void addVertex(CovNode a)
    {
        super.addVertex(a);
    }
    
    public void addEdge(CovLink b, CovNode a1, CovNode a2)
    {
        super.addEdge(b, a1, a2);
    }
    
    @Override
    public void addVertex(AbsVertex v)
    {
        try
        {
            addVertex((CovNode) v);
        }
        catch (Exception e)
        {
            e.printStackTrace();
        }
    }

    @Override
    public void addEdge(AbsEdge a, AbsVertex v1, AbsVertex v2)
    {
        try
        {
            addEdge((CovLink) a, (CovNode) v1, (CovNode) v2);
        }
        catch (Exception e)
        {
            e.printStackTrace();
        }
    }
}

File: scala/src/covariance/informatics/CovNetwork.scala
package covariance.informatics

import covariance.abs.AbsEdge
import covariance.abs.AbsGraph
import covariance.abs.AbsVertex

class CovNetwork(name:String) extends AbsGraph("Network " + name) {
  
  def addVertex(a: CovNode): Unit = {
    super.addVertex(a)
  }
  def addEdge(b: CovLink, a1: CovNode, a2: CovNode): Unit = {
    super.addEdge(b, a1, a2)
  }

  override def addVertex(v: AbsVertex): Unit = {
    try {
      addVertex(v.asInstanceOf[CovNode])
    } catch {
      case e: Exception =>
        e.printStackTrace()

    }
  }

  override def addEdge(a: AbsEdge, v1: AbsVertex, v2: AbsVertex): Unit = {
    try {
      addEdge(a.asInstanceOf[CovLink], v1.asInstanceOf[CovNode], v2.asInstanceOf[CovNode])
    } catch {
      case e: Exception =>
        e.printStackTrace()

    }
  }

}
